packages/internals/support/src/entities/component.js

```diff
@@ -64,6 +51,25 @@ class Component extends Entity {
       get: variantsGet,
       set: variantsSet
     });
+
+    Object.defineProperty(this, 'config', {
+      enumerable: true,
+      get: configGet,
+      set: configSet
+    });
+  }
+
+  get(path, fallback) {
+    fallback = get(this.config, path, fallback);
+    const result = get(_data.get(this), path, fallback);
+    return isObjectLike(result) ? cloneDeep(result) : result;
+  }
+
+  set(path, value) {
+    assert.string(path, 'Component.set - `path` argument must be a string [path-invalid]');
+    // TODO: extract shared caching code with config: this.removeFromCache(path);
+    set(_data.get(this), path, cloneDeep(value));
```

Comment:
The tests do contain string values.
I'm pretty sure it checks that itself (though I'd also copied that check in the getter, I should at least be consistent!)
I've also checked it out on runkit, seems to just pass primitives back - though it just seems to have gone down. I think I broke it 😇